airbyte-cdk/java/airbyte-cdk/dependencies/src/main/kotlin/io/airbyte/commons/concurrency/CompletableFutures.kt

```diff
@@ -24,9 +24,7 @@ object CompletableFutures {
         val result = CompletableFuture<List<Either<out Exception, Result>>>()
         val size = futures.size
         val counter = AtomicInteger()
-        val results =
-            java.lang.reflect.Array.newInstance(Either::class.java, size)
-                as Array<Either<Exception, Result>>
+        val results = ArrayList<Either<out Exception, Result>>(size)
```

Comment:
Does this _need_ to be an `ArrayList`? If not it should probably be `mutableListOf<Either<out Exception, Result>>()`

Reply:
actually, neither ArrayList nor mutableListOf work. I have to stick to the array apparently :(